### ISSUE TITLE
crashes after update to 2.1.0.9

### ISSUE THREAD

**Issue by cosminadrianpopescu** [sailfishos/sailfish-browser#554]
I've just updated the sailfish os to the latest version (on Friday) and since then I cannot start the browser:

```
[nemo@Sailfish ~]$ sailfish-browser
[D] unknown:0 - Using Wayland-EGL
greHome from GRE_HOME:/usr/bin
libxul.so is not found, in /usr/bin/libxul.so
Created LOG for EmbedLite
Segmentation fault
[nemo@Sailfish ~]$
```

**Comment by schmittlauch** [comment]
Have you tried to open a link to another site from some app (e.g. Jolla Store, mail client)? There are some sitest that crash the browser, but you can close those tabs if you manage to open the browser with another site it doesn't crash on.

**Comment by schmittlauch** [comment]
I also have the feeling that more pages crash the browser since upgrading to 2.1.0.9 and I'm not the only one according to [this TJC diskussion](https://together.jolla.com/question/156676/2109-browser-starts-and-hangs/ ).
But when I tried to find a website to reproducibly crash the browser I got to #555 which also crashes the browser in 2.0.5.6 .

Another candidate for often, but not always crashing the browser (probably since 2.1) is mobile twitter. The problem there is that they changed their whole mobile frontend stack around the time sfos 2.1 came out. So , having only one phone, I'm not really able to tell whether it's caused by a regression introduced into sailfish-browser after 2.0.5.6 or it's caused by the new twitter frontend and also crashing older versions of the browser.

**Comment by cosminadrianpopescu** [comment]
the browser doesn't crash. Please read the initial post carefully. It doesn't stary. It's a dependency issue

**Comment by schmittlauch** [comment]
So it didn't even start once, tried to load, then crashed ant didn't want to start after that? Sorry, then I misinterpreted the issue.

**Comment by cosminadrianpopescu** [comment]
Yes. That's the problem. After upgrading to 2.1.0.9 I've had a lot of dependency issues. I've saw that one of the points of 2.1.0.9 is upgrading qt to 5.6. I've had more softwares which were complayining about qute libraries from 5.5. Now I am on 2.0.5 and not sure if I should upgrade again on 2.1.0.9.

**Comment by schmittlauch** [comment]
This sounds more like a general update problem of SailfishOS, not something specific to sailfish-browser. Did you make sure to have a well-balanced filesystem with enough free space before updating?

**Comment by cosminadrianpopescu** [comment]
yes, but right now i've reset my phone anyway, so i suppose you can close this ticket.

if you don't know of any dependency issues with the last update, please let me know and close the ticket.

in your oppinion, is safe to upgrade again to 2.1? rght now, as i've mentioned, i am using 2.0.5

**Comment by schmittlauch** [comment]
I don't know if #558 is only related to my device, but otherwise I see no reason no to upgrade to 2.1.0.9 if you make sure to have enough free space and a well-balanced btrfs.

**Comment by rainemak** [comment]
I'm sorry to hear that you're facing issues with 2.1.0. Let's try to find a solution for your problem. First of all, I'd like to verify that you have not installed Sailfish Browser from open repository or you have open repositories enabled that contain browser packages. If that part is good, we could try out something....

Do you guys have developer mode enabled? If so could you test by any chance following:

```
close browser
cp -r /home/nemo/.mozilla /home/nemo/.mozilla-bak
rm -rf /home/nemo/.mozilla/mozembed/startupCache
restart browser
```
Password caches etc are now on the .mozilla-bak so that you can recover them if needed.

If that does not help:
```
close browser
rm -rf /home/nemo/.mozilla
restart browser
```
If the last resort ``rm -rf /home/nemo/.mozilla``` does help then you have lost e.g. all saved password and caches. Browsing history, bookmarks, and session history will be restored upon browser restart.

You can restore to the original state:
```
close browser
rm -rf /home/nemo/.mozilla
mv /home/nemo/.mozilla-bak /home/nemo/.mozilla
restart browser
```

Let me know if this help.

**Comment by schmittlauch** [comment]
@rainemak I didn't have these issues, but I am struggling with #558. And @cosminadrianpopescu has downgraded to the SFOS version before to get rid of his issue. So I think this can be closed for now and may be reopened if @cosminadrianpopescu can reproduce it again.

I'll try whether your suggestions fix #558 though.

**Comment by cosminadrianpopescu** [comment]
@rainemak: as specified earlier, I reset my phone, so I cannot test anymore. However, as stated, it was a dependency issue. Nothing to do with the cache. Please read all the posts here. 

Thank you.
